app/WebKnossosModule.scala

```diff
@@ -22,6 +23,7 @@ class WebKnossosModule extends AbstractModule {
     bind(classOf[UserCache]).asEagerSingleton()
     bind(classOf[AnnotationStore]).asEagerSingleton()
     bind(classOf[DataSetService]).asEagerSingleton()
+    bind(classOf[DataSourceService]).asEagerSingleton()
```

Comment:
Do you think it would be possible to extract the part of DataSourceService’s functionality that needs it to be singleton to another, smaller, service? I feel like it already has a lot of tasks that may perform better if the injector can create multiple instances dynamically. (things to double-check there: the classes the singletons are injected into don’t automatically become singletons too, right?). This may be a good strategy for other services as well, in the future.

Reply:
I just tested it and it works :tada: 
> the classes the singletons are injected into don’t automatically become singletons too

No, they can still have multiple instances